controllers/tenant/namespaces.go

```diff
@@ -73,6 +73,25 @@ func (r *Manager) syncNamespaceMetadata(ctx context.Context, namespace string, t
 				}
 			}
 
+			if tnt.Spec.NamespaceOptions != nil && len(tnt.Spec.NamespaceOptions.AdditionalMetadataList) > 0 {
+				for _, additionalMetadata := range tnt.Spec.NamespaceOptions.AdditionalMetadataList {
+					var namespaceSelectedBySelector bool
+
+					namespaceSelectedBySelector, err = utils.IsNamespaceSelectedBySelector(ns, additionalMetadata.NamespaceSelector)
+					if err != nil {
+						return err
+					}
+
+					if !namespaceSelectedBySelector {
+						continue
+					}
+
+					for k, v := range additionalMetadata.Labels {
+						labels[k] = v
+					}
```

Comment:
Add annotations as well

```suggestion
					for k, v := range additionalMetadata.Annotations{
						annotations[k] = v
					}
			}
```

Reply:
Added